lib/galaxy/jobs/runners/tes.py

```
@@ -0,0 +1,547 @@
+"""
+Job control via TES.
+"""
+import logging
+import os
+from re import escape, findall
+
+import requests
+
+from galaxy import model
+from galaxy.jobs import JobWrapper
+from galaxy.jobs.command_factory import build_command
+from galaxy.jobs.runners import (
+    AsynchronousJobRunner,
+    AsynchronousJobState
+)
+from galaxy.util import asbool
+
+log = logging.getLogger(__name__)
+
+__all__ = ('TESJobRunner', )
+
+GENERIC_REMOTE_ERROR = "Failed to communicate with remote job server."
+FAILED_REMOTE_ERROR = "Remote job server indicated a problem running or monitoring the job."
+LOST_REMOTE_ERROR = "Remote job server could not determine the job's state."
+
+DEFAULT_GALAXY_URL = "http://localhost:8080/"
+
+
+class TESJobState(AsynchronousJobState):
+    def __init__(self, **kwargs):
+        """
+        Encapsulates state related to a job.
+        """
+        super(TESJobState, self).__init__(**kwargs)
+        self.user_log = None
+        self.user_log_size = 0
+        self.cleanup_file_attributes = ['output_file', 'error_file', 'exit_code_file']
+
+
+class TESJobRunner(AsynchronousJobRunner):
+    """
+    Job runner backed by a finite pool of worker threads. FIFO scheduling
+    """
+    runner_name = "TESJobRunner"
+
+    def __init__(self, app, nworkers, **kwargs):
+        """
+            Initialize this job runner and start the monitor thread
+        """
+        super(TESJobRunner, self).__init__(app, nworkers, **kwargs)
+        self.container_workdir = "/tmp"
+        if(hasattr(app.config, "galaxy_infrastructure_url")):
+            self.galaxy_url = f"{app.config.galaxy_infrastructure_url.rstrip('/')}/"
+        else:
+            self.galaxy_url = DEFAULT_GALAXY_URL
```

I think this should raise an exception, I think defaulting to `http://localhost:8080/` might be harder to debug than raising an exception here.

---

Sure, I can raise an exception here.

---

Fixed